src/qililab/instruments/quantum_machines/quantum_machines_cluster.py

```diff
@@ -354,52 +351,68 @@ def set_parameter_of_bus(self, bus: str, parameter: Parameter, value: float | st
-                self._config["octaves"][octave_name]["RF_outputs"][out_port]["LO_frequency"] = lo_frequency
+                # 2) Change config if exists:
+                if self._config_created:
+                    self._config["octaves"][octave_name]["RF_outputs"][out_port]["LO_frequency"] = lo_frequency
+                # 3) Change instrument if connected:
+                if self._is_connected_to_qm:
+                    self._qm.octave.set_lo_frequency(element=bus, lo_frequency=lo_frequency)
 
                 if in_port is not None:
                     settings_octave_rf_input = next(
                         rf_input for rf_input in settings_octave["rf_inputs"] if rf_input["port"] == in_port
                     )
+                    # 1) Change settings:
                     settings_octave_rf_input["lo_frequency"] = lo_frequency
-                    self._config["octaves"][octave_name]["RF_inputs"][in_port]["LO_frequency"] = lo_frequency
+                    # 2) Change config if exists:
+                    if self._config_created:
+                        self._config["octaves"][octave_name]["RF_inputs"][in_port]["LO_frequency"] = lo_frequency
                 return
+
             if parameter == Parameter.GAIN:
                 gain_in_db = float(value)
-                self._qm.octave.set_rf_output_gain(element=bus, gain_in_db=gain_in_db)
+                # 1) Change settings:
                 settings_octave_rf_output["gain"] = gain_in_db
-                self._config["octaves"][octave_name]["RF_outputs"][out_port]["gain"] = gain_in_db
+                # 2) Change config if exists:
+                if self._config_created:
+                    self._config["octaves"][octave_name]["RF_outputs"][out_port]["gain"] = gain_in_db
+                # 3) Change instrument if connected:
+                if self._is_connected_to_qm:
+                    self._qm.octave.set_rf_output_gain(element=bus, gain_in_db=gain_in_db)
                 return
+
         if parameter == Parameter.IF:
             intermediate_frequency = float(value)
-            self._qm.set_intermediate_frequency(element=bus, freq=intermediate_frequency)
+            # 1) Change settings:
             element["intermediate_frequency"] = intermediate_frequency
-            self._config["elements"][bus]["intermediate_frequency"] = intermediate_frequency
-
-            if f"mixer_{bus}" in self._config["mixers"]:
+            # 2) Change config if exists:
+            if self._config_created:
+                self._config["elements"][bus]["intermediate_frequency"] = intermediate_frequency
+            if self._config_created and f"mixer_{bus}" in self._config["mixers"]:
                 self._config["mixers"][f"mixer_{bus}"][0]["intermediate_frequency"] = intermediate_frequency
+            # 3) Change instrument if connected:
+            if self._is_connected_to_qm:
+                self._qm.set_intermediate_frequency(element=bus, freq=intermediate_frequency)
```

Comment:
Bear in mind that this will be changed in the future into job.set_intermediat_frequency (I will work in this basis for qua 1.2.0)

Reply:
Cool! Tell me something if you need a hand with the implementation then!

Reply:
But I guess this is not problem for this current code, no? 
Resolving then!